lib/chef/audit/audit_reporter.rb

```diff
@@ -116,8 +118,10 @@ def post_auditing_data(error = nil)
         Chef::Log.debug("Sending audit report (run-id: #{audit_data.run_id})")
         run_data = audit_data.to_hash
 
-        if error
-          run_data[:error] = "#{error.class.to_s}: #{error.message}\n#{error.backtrace.join("\n")}"
+        if @exception || error
+          errors = [@exception, error].uniq.compact
+          errors_messages = errors.map  { |err| "#{err.class.to_s}: #{err.message}\n#{err.backtrace.join("\n")}" }
```

Comment:
You can leverage your `error_message_for_run_data` method you defined below in this map.  I'm pretty sure ruby allows you to do something like `errors.map &error_message_for_run_data` but someone with more ruby knowledge could chime in on that.